rewrite-core/src/main/java/org/openrewrite/SourceFileWithTypeReferences.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite;
+
+import org.openrewrite.internal.JavaTypeReferences;
+
+public interface SourceFileWithTypeReferences {
```

Comment:
Not sure if this is over interfacing it, currently it's only about XML documents. Do we see any future use cases for other types of SourceFiles

Reply:
After talking to Knut: Both Spring and Quarkus have an application config file that could be written in YAML and could contain type references, so there are definitely use cases outside of Xml.